app/models/user.rb

```diff
@@ -11,4 +11,8 @@ class User < ApplicationRecord
 
   validates :full_name, presence: true
   validates :gender, presence: true, inclusion: { in: genders.keys }
+
+  def chats
+    Chat.where(user_a_id: id) + Chat.where(user_b_id: id)
```

Comment:
Maybe you can combine this two queries, so you make only one request to DB.. WDYT?

Reply:
Yes, I think is the optimal way to do it!